test/extended/dns/dns.go

```diff
@@ -130,21 +148,13 @@ func validateDNSResults(f *e2e.Framework, pod *api.Pod, fileNames sets.String, e
 		e2e.Failf("Failed to create %s pod: %v", pod.Name, err)
 	}
 
-	Expect(f.WaitForPodRunning(pod.Name)).To(BeNil())
-	Expect(wait.Poll(2*time.Second, 5*time.Minute, func() (bool, error) {
-		pod, err := podClient.Get(pod.Name)
-		if err != nil {
-			return false, err
-		}
-		switch pod.Status.Phase {
-		case api.PodSucceeded:
-			return true, nil
-		case api.PodFailed:
-			return false, fmt.Errorf("pod failed")
-		default:
-			return false, nil
-		}
-	})).To(BeNil())
+	w, err := f.Client.Pods(f.Namespace.Name).Watch(api.SingleObject(api.ObjectMeta{Name: pod.Name}))
```

Comment:
This is the problem upstream, the Watch doesn't have ResourveVersion set which causes us missing certain events, can you add it?